### DISCUSSION TITLE
New Module for smartwatches display

### DISCUSSION THREAD

**Discussion by RomaricKc1** [osmandapp/OsmAnd#18502] [Ideas]
Hello,

I recently installed the application on my Galaxy Watch 5 44mm, and it's been good.
From #1433, I decided to work on the layout. The buttons are more accessible and most of the menus have a visible option/back button to ease navigation. 

I added a new module to the project and edited the `res` files, and some of the `src` files (just for the layout). I didn't make changes to the main `OsmAnd` since this is for regular-sized Android devices.

 I want to know if it would be a welcome change or not. I am aware that what I propose is not built for Wear Os devices. Some of the layouts are still not optimized for these display sizes.

A couple of things that would be nice that I haven't figured out yet:

- ~~Keep the map activity running even when dismissed with a swipe to the right~~ swipe to dismiss disabled
- Some invisible back buttons, tho I relocated them and the area is clickable
- Optimize certain views launched from the code (like the Last Crashed --partially optimized, speed camera warning, and others)

PS: I cannot run the modified layout version on my watch yet, it defaults to an arch that is not supported by the watch. Any help would be appreciated.

Edit: I've changed the build variant to fullopenglarmv7 and it works.
Dialog buttons fixed
Edit2: new UI navigation improvements plus **swipe to dismiss** disabled. 

Edit3: Finally got a working version and made this #21425.

I've got these with the Emulator on Android Studio size: Wear Os Large:
| Picture  | picture|
|-----------------|:-------------|
| ![pic3](https://github.com/osmandapp/OsmAnd/assets/45500198/cbfc208f-c330-4788-9802-36f768d3d6fd) | ![pic4](https://github.com/osmandapp/OsmAnd/assets/45500198/598b630d-11cc-4022-a4a9-24d1fa5f36b5)  |
| ![pic5](https://github.com/osmandapp/OsmAnd/assets/45500198/97427ebf-416a-4a5f-87ef-f5aeb36a6dca) | ![pic14](https://github.com/osmandapp/OsmAnd/assets/45500198/164ae5d1-2de9-439f-95be-04890b9d70e7) |
| ![pic14](https://github.com/osmandapp/OsmAnd/assets/45500198/164ae5d1-2de9-439f-95be-04890b9d70e7) | ![pic13](https://github.com/osmandapp/OsmAnd/assets/45500198/1699e666-3295-44d7-bb77-f128e218f836)  |
| ![pic12](https://github.com/osmandapp/OsmAnd/assets/45500198/69a55473-d3b7-4d70-b1fc-b75f90a37f70) | ![pic11](https://github.com/osmandapp/OsmAnd/assets/45500198/8eb64108-96c7-4879-8276-4b7eea97cb4f)  |

**Comment by onyx666** [comment]
how can i get this module ?

**Reply by RomaricKc1** [reply]
I have a forked draft where I modified the layout. I haven't received any feedback from the maintainers here (in this discussion) whether it's something they would like. 
With the new changes made to the source code (upstream), I tried a couple of times  to make a release with the new layout but couldn't even build the application anymore. Some compilation and testing errors (even on a vanilla clone). So I kind of put it in stall. I don't know the project that much to fix the errors.

Though I have the full working version on my Galaxy Watch 5 (44mm) I believe it might look weird on smaller screen I haven't tested. But yeah, based on outdated code (upstream) I had cloned back then.
In conclusion, the module isn't published anywhere. I don't think I should put it in a repo with the old source codes.
I might take a look again in the repo if there were so bugs fixes or maybe I was just doing it wrong.

**Comment by vicb** [comment]
That would be super nice to be able to run osmand on WearOS watches!

**Comment by artemigor** [comment]
+1, interesting to use

**Comment by danerrickcuba** [comment]
hello...Any progress?

**Reply by RomaricKc1** [reply]
Hello
I made this pr #21425. Waiting to hear from the maintainers.
